src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3392,5 +3412,49 @@ private void EmitBox(TypeSymbol type, SyntaxNode syntaxNode)
             _builder.EmitOpCode(ILOpCode.Box);
             EmitSymbolToken(type, syntaxNode);
         }
+
+        private void EmitCalli(BoundFunctionPointerInvocation ptrInvocation, UseKind useKind)
+        {
+            FunctionPointerMethodSymbol method = ptrInvocation.FunctionPointer.Signature;
+            EmitArguments(ptrInvocation.Arguments, method.Parameters, ptrInvocation.ArgumentRefKindsOpt);
+            var stackBehavior = GetCallStackBehavior(ptrInvocation.FunctionPointer.Signature, ptrInvocation.Arguments);
+            EmitExpression(ptrInvocation.InvokedExpression, used: true);
```

Comment:
>EmitExpression(ptrInvocation.InvokedExpression, used: true); [](start = 12, length = 60)

It looks like this is going to violate an evaluation order. Do we have a test asserting the evaluation order, i.e. the target is evaluated before the arguments?

Reply:
Yes, we have multiple. The LocalRewriter change is responsible for spilling the InvokedExpression if necessary to ensure evaluation happens in the correct order.

---
In reply to: [382381636](https://github.com/dotnet/roslyn/pull/41794#discussion_r382381636) [](ancestors = 382381636)